### PR TITLE
Reindex: Fold "with all deps" test project into reindex

### PR DESCRIPTION
**Pull request by nik9000**
This folds the `:qa:smoke-test-reindex-with-all-modules` project into
`:modules:reindex` by declaring the reindex's integration testing
cluster requires the `parent-join` and `lang-painless` plugins and then
moving all of the integration tests that depended on parent-join and
painless into reindex.

It saves us one cluster start up during the build at the cost of a
little of the reindex module's "purity". Since the reindex module *does*
have unit tests that test scripting without painless I'm fairly ok with
that.

### CODE REVIEW

File: qa/smoke-test-reindex-with-all-modules/src/test/resources/rest-api-spec/test/reindex/20_broken.yml [hunk 1]
```diff
@@ -1,24 +0,0 @@
...
-        refresh: true
-        body:
-          source:
-            index: twitter
-          dest:
-            index: new_twitter
-          script:
-            lang: painless
-            source: syntax errors are fun!
-  - match: {error.reason: 'compile error'}
```
Comment (nik9000) [inline qa/smoke-test-reindex-with-all-modules/src/test/resources/rest-api-spec/test/reindex/20_broken.yml]: I moved this into the scripting file.

File: qa/smoke-test-reindex-with-all-modules/src/test/resources/rest-api-spec/test/reindex/30_timeout.yml [hunk 2]
```diff
@@ -1,33 +0,0 @@
----
-"Timeout":
-  - skip:
-      version: "all"
-      reason:  painless doesn't support thread.sleep so we need to come up with a better way
```
Comment (nik9000) [inline qa/smoke-test-reindex-with-all-modules/src/test/resources/rest-api-spec/test/reindex/30_timeout.yml]: We haven't been running this for *forever* and we've been ok so I just dropped it.

File: qa/smoke-test-reindex-with-all-modules/src/test/resources/rest-api-spec/test/update_by_query/20_broken.yml [hunk 3]
```diff
@@ -1,21 +0,0 @@
----
-"Totally broken scripts report the error properly":
```
Comment (nik9000) [inline qa/smoke-test-reindex-with-all-modules/src/test/resources/rest-api-spec/test/update_by_query/20_broken.yml]: I moved this into the scripting file.

File: qa/smoke-test-reindex-with-all-modules/src/test/resources/rest-api-spec/test/update_by_query/30_timeout.yml [hunk 4]
```diff
@@ -1,30 +0,0 @@
----
-"Timeout":
```
Comment (nik9000) [inline qa/smoke-test-reindex-with-all-modules/src/test/resources/rest-api-spec/test/update_by_query/30_timeout.yml]: We haven't been running this in *forever* so I just dropped it.

**Comment by nik9000** [comment]
The build failure was a timeout waiting for x-pack.

@elasticmachine, retest this please.

**Comment by nik9000** [comment]
Sigh. @elasticmachine, retest this please

**Review by jasontedor — APPROVED** [review]
LGTM. We might revisit this depending on if we come to a conclusion about `integ-test-zip`.

**Comment by nik9000** [comment]
> LGTM. We might revisit this depending on if we come to a conclusion about integ-test-zip.

Thanks! I agree. We might switch around the gradle bit a bit if we make a decision on integ-test-zip but I think this is a step in the right direction in the mean time.

I've merge and am backporting now.

**Pull request merged**